### PR TITLE
Unification of BreadcrumbItem and BreadcrumbDropDownItem

### PR DESCRIPTION
**Pull request by RBrid**
BreadcrumbItem and BreadcrumbDropDownItem classes are merged into a single BreadcrumbItem object. Breadcrumb.DropDownItemTemplate property is removed too.

BreadcrumbDropDown, BreadcrumbDropDownElementFactory, BreadcrumbDropDownAutomationPeer are merged into BreadCrumbItem classes.
The concept of drop-down-item is no longer - we reserve that term for a future release that could have dropdowns shown through the chevrons. Instead we have ellipsis-drop-down-item. Those are specifically for the flyout shown through the ellipsis item. Thus 'EllipsisDropDownItem' appears throughout the code.

Did a few renames in the process for more consistent names.

These changes include a workaround for an existing crashing bug when attempting to show the ellipsis item's flyout. The workaround is in BreadcrumbElementFactory::RecycleElementCore.

### CODE REVIEW

File: dev/Breadcrumb/Breadcrumb.xaml [hunk 1]
```diff
@@ -306,13 +293,11 @@
...
                 <ControlTemplate TargetType="local:Breadcrumb">
-                    <local:ItemsRepeater x:Name="PART_BreadcrumbItemsRepeater" />
+                    <local:ItemsRepeater x:Name="PART_ItemsRepeater" />
                 </ControlTemplate>
             </Setter.Value>
         </Setter>
     </Style>
 
-    <Style TargetType="local:BreadcrumbDropDownItem" BasedOn="{StaticResource BreadcrumbDropDownItemStyle}"/>
     <Style TargetType="local:BreadcrumbItem" BasedOn="{StaticResource BreadcrumbItemStyle}"/>
```
Comment (ranjeshj) [inline dev/Breadcrumb/Breadcrumb.xaml:302]: nit:DefaultBreadcrumbItemStyle is the pattern we use.

File: dev/Breadcrumb/BreadcrumbItem.cpp [hunk 2]
```diff
@@ -70,56 +93,62 @@ void BreadcrumbItem::RevokePartsListeners()
...
+    {
+        RevokePartsListeners();
 
-    m_breadcrumbItemButton.set(GetTemplateChildT<winrt::Button>(s_breadcrumbItemButtonPartName, controlProtected));
+        winrt::IControlProtected controlProtected{ *this };
 
-    if (const auto& breadcrumbItemButton = m_breadcrumbItemButton.get())
-    {
-        m_breadcrumbItemButtonLoadedRevoker = breadcrumbItemButton.Loaded(winrt::auto_revoke, { this, &BreadcrumbItem::OnLoadedEvent });
+        m_button.set(GetTemplateChildT<winrt::Button>(s_itemButtonPartName, controlProtected));
```
Comment (ranjeshj) [inline dev/Breadcrumb/BreadcrumbItem.cpp:108]: >m_button [](start = 8, length = 8)

nit:Could be buttonbase ?

**Review by ranjeshj — APPROVED** [review]
:shipit:

File: dev/Breadcrumb/BreadcrumbItem.cpp (hunk 2 above)
Comment (RBrid) [reply dev/Breadcrumb/BreadcrumbItem.cpp:108]: Nope. ButtonBase.Flyout does not exist.  Used @ line 578.

**Pull request merged**